test/testSheet.cc

```diff
@@ -0,0 +1,237 @@
+    EXPECT_TRUE(onResizeHandle(100 + w, 100 + h));
+    
+    // Test top-right resize handle
+    EXPECT_TRUE(onResizeHandle(100 + w, 100 - h));
+    
+    // Test bottom-left resize handle (when showRavel is false)
+    showRavel = false;
+    EXPECT_TRUE(onResizeHandle(100 - w, 100 + h));
+    
+    // Test points not on resize handle
+    EXPECT_FALSE(onResizeHandle(100, 100));
+  }
+
+  TEST_F(SheetTest, inItem)
+  {
+    moveTo(100, 100);
+    iWidth(200);
+    iHeight(150);
+    
+    // Test point inside item
+    EXPECT_TRUE(inItem(100, 100));
+    
+    // Test point outside item
+    EXPECT_FALSE(inItem(0, 0));
+    EXPECT_FALSE(inItem(300, 300));
+  }
+
+  TEST_F(SheetTest, clickType)
+  {
+    moveTo(100, 100);
+    iWidth(200);
+    iHeight(150);
+    double z = zoomFactor();
+    double w = 0.5 * 200 * z;
+    double h = 0.5 * 150 * z;
+    
+    // Test resize handle
+    EXPECT_EQ(ClickType::onResize, clickType(100 + w, 100 + h));
+    
+    // Test inside item
+    EXPECT_EQ(ClickType::inItem, clickType(100, 100));
+    
+    // Test on item border
+    EXPECT_EQ(ClickType::onItem, clickType(100 + w * 0.9, 100 + h * 0.9));
+    
+    // Test outside item
+    EXPECT_EQ(ClickType::outside, clickType(0, 0));
+  }
+
+  TEST_F(SheetTest, contains)
+  {
+    moveTo(100, 100);
+    iWidth(200);
+    iHeight(150);
+    
+    // Test point inside item
+    EXPECT_TRUE(contains(100, 100));
+    
+    // Test point outside item
+    EXPECT_FALSE(contains(0, 0));
+  }
+
+  TEST_F(SheetTest, scrollUpDown)
+  {
+    // Initial state - no scrolling possible without data
+    EXPECT_FALSE(scrollUp());
+    EXPECT_FALSE(scrollDown());
+    // add a rank 3 tensor, which is "scrollable"
+    minsky::minsky().canvas.addVariable("3dTensor",VariableType::parameter);
+    auto& param=*minsky::minsky().canvas.itemFocus->variableCast();
+    param.init("rand(3,3,3)");
+    minsky::minsky().canvas.addSheet();
+    auto& sheet=dynamic_cast<Sheet&>(*minsky::minsky().canvas.itemFocus);
+    minsky::minsky().model->addWire(param, sheet, 0);
+    minsky::minsky().reset();
+    EXPECT_TRUE(sheet.scrollUp());
+    EXPECT_TRUE(sheet.scrollDown());
+  }
+
+  TEST_F(SheetTest, onKeyPress)
+  {
+    // Test arrow key handling
+    // Up arrow (0xff52)
+    EXPECT_FALSE(onKeyPress(0xff52, "", 0));
+    
+    // Down arrow (0xff54)
+    EXPECT_FALSE(onKeyPress(0xff54, "", 0));
+    
+    // Right arrow (0xff53)
+    EXPECT_FALSE(onKeyPress(0xff53, "", 0));
+    
+    // Left arrow (0xff51)
+    EXPECT_FALSE(onKeyPress(0xff51, "", 0));
+    
+    // Other key
+    EXPECT_FALSE(onKeyPress(0x0041, "A", 0));
+  }
+
+  TEST_F(SheetTest, exportAsCSVWithoutValue)
+  {
+    // Attempting to export without a value should throw an error
+    EXPECT_THROW(exportAsCSV("/tmp/test.csv", false), std::exception);
+  }
+
+  TEST_F(SheetTest, setSliceIndicator)
+  {
+    // With no value, setSliceIndicator should handle gracefully
+    setSliceIndicator();
```

Comment:
Need to add a rank 3 tensor, then check that sliceIndicator is setto something useful.